test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+
+#include "../util/IndexTestHelpers.h"
+#include "./../../src/global/ValueId.h"
+#include "./../../src/util/GeoSparqlHelpers.h"
+#include "engine/ExportQueryExecutionTrees.h"
+#include "engine/IndexScan.h"
+#include "engine/Join.h"
+#include "engine/QueryExecutionTree.h"
+#include "engine/SpatialJoin.h"
+#include "parser/data/Variable.h"
+
+using namespace ad_utility::testing;
+
+// helper function in debugging for outputting stuff
+void print_vecs(std::vector<std::vector<std::string>> vec) {
+  for (size_t i = 0; i < vec.size(); i++) {
+    for (size_t k = 0; k < vec.at(i).size(); k++) {
+      std::cerr << vec.at(i).at(k) << " ";
+    }
+    std::cerr << std::endl;
+  }
+}
+
+// helper function in debugging for outputting stuff
+void print_vec(std::vector<std::string> vec) {
+  for (size_t i = 0; i < vec.size(); i++) {
+    std::cerr << vec.at(i) << std::endl;
+  }
+}
```

Comment:
Can't you use googletest's builting facilities `PrintTo[String]` here?
They should have capabilities for `vector` and `string`.

Reply:
this was just needed for debugging during the development. I completely removed them